vector/src/main/java/im/vector/app/features/pin/lockscreen/biometrics/BiometricHelper.kt

```diff
@@ -174,16 +180,18 @@ class BiometricHelper @Inject constructor(
                 when (val exception = result.exceptionOrNull()) {
                     null -> result.getOrNull()?.let { emit(it) }
                     else -> {
-                        // Exception found, stop collecting, throw it and remove the prompt reference
+                        // Exception found:
+                        // 1. Stop collecting.
+                        // 2. Remove the system key if we were creating it.
+                        // 3. Throw the exception and remove the prompt reference
+                        if (!checkSystemKeyExists) {
+                            lockScreenKeyRepository.deleteSystemKey()
+                        }
                         prompt = null
                         throw exception
                     }
                 }
             }
-            // Generates the system key on successful authentication
-            if (buildVersionSdkIntProvider.get() >= Build.VERSION_CODES.M) {
-                lockScreenKeyRepository.ensureSystemKey()
-            }
```

Comment:
Since we started using `CryptoObject`, the system key is created when that `CryptoObject` is instantiated. Instead of ensuring the system key is created on success, make sure it's removed on error if we were on the process of enabling biometric auth.